wiring/inc/spark_wiring_system.h

```diff
@@ -148,8 +148,18 @@ class SystemClass {
 
     inline int enableFeature(HAL_Feature feature)
     {
+        /* PRE-ENABLE TASKS */
+        bool init_sram = false;
+        // Initialize SRAM on first use of retained memory
+        if (feature==FEATURE_RETAINED_MEMORY && !HAL_Feature_Get(FEATURE_RETAINED_MEMORY)) {
+            init_sram = true;
+        }
+
+        /* ENABLE FEATURE */
         int result = HAL_Feature_Set(feature, true);
-        if (feature==FEATURE_RETAINED_MEMORY && !HAL_Feature_Get(FEATURE_WARM_START)) {
+
+        /* POST-ENABLE TASKS */
+        if (feature==FEATURE_RETAINED_MEMORY && init_sram) {
```

Comment:
I'm wondering if we should always initialize the backup RAM in HAL_Core_Config() or some other early startup function, and only if the backup regular is off.